juju/juju
# [JUJU-4211] Remove support for podspec charms

Operator style (podspec) charms are no longer supported in Juju4. This PR removes support for deploying them.
The relevant facades and workers are removed totally (eg caasoperatorprovisioner). Other facades like the caasunitprovisioner has had a haircut to remove methods which were only used by the old workers. And the k8s provider code has had a lot of now obsolete methods and tests removed. 
There were 2 firewallers - the legacy one is removed "sidecar" variant renamed to take its place.

## QA steps

bootstrap to microk8s
deploy snappass-test
scale-application snappass-test 2
deploy mysql-k8s

run status --storage to check that storage is created for mysql
